pkg/keymanagementprovider/azurekeyvault/provider_test.go

```diff
@@ -373,13 +594,165 @@ func TestGetCertificates(t *testing.T) {
 
 // TestGetKeys tests the GetKeys function
 func TestGetKeys(t *testing.T) {
-	keyID := azkeys.ID("https://testkv.vault.azure.net/keys/key1")
+	keyID := azkeys.ID("https://testkv.vault.azure.net/keys/key1/c1f03df1113d460491d970737dfdc35d")
+	keyIDLatest := azkeys.ID("https://testkv.vault.azure.net/keys/key1/8f2e5a13c4b74960d7a8e2f1c0d6b3a9")
+	keyCreated := time.Now()
+	keyCreatedLatest := time.Now().Add(1 * time.Minute)
 	keyTY := azkeys.JSONWebKeyTypeRSA
 	testCases := []struct {
-		name            string
-		mockKeyKVClient *MockKeyKVClient
-		expectedErr     bool
+		name                string
+		versionHistoryLimit int
+		mockKeyKVClient     *MockKeyKVClient
+		expectedErr         bool
 	}{
+		{
+			name:                "Key enabled with multiple versions",
+			versionHistoryLimit: 3,
```

Comment:
Given the method now have additional logic around pager, sorting etc, the test should validate the length of the response and which versions are returned. Suggestion to update the current validation. If there is a error thrown, we should also validate if its the expected error msg.

```
_, _, err := provider.GetKeys(context.Background())
```

Reply:
Hey @susanshi I addressed this in a different branch as part of a refactor. Do you want those changes pulled into this PR? Or is it okay to include them in a follow up PR with the refactoring?